contracts/rootchain/RootChain.sol

```diff
@@ -210,8 +210,11 @@ contract RootChain is Ownable {
             } else
                 state = depositExits[depositNonce_].state;
 
-            require(state == ExitState.NonExistent || state == ExitState.Challenged, "inputs are being exited or have been finalized");
+            if (state != ExitState.NonExistent && state != ExitState.Challenged)
```

Comment:
I believe having all the `require` statements at the top level function is more clear rather than potentially erroring in the children. The top level function should reflect all possible flows of execution including where things can revert